pkg/cmd/repo/create/create.go

```diff
@@ -404,6 +411,114 @@ func createFromScratch(opts *CreateOptions) error {
 		} else if err := cloneWithRetry(opts, remoteURL, templateRepoMainBranch); err != nil {
 			return err
 		}
+
+	}
+
+	return nil
+}
+
+// create new repo on remote host from template repo
+func createFromTemplate(opts *CreateOptions) error {
+	// this method is only for interactive mode
+	if !opts.Interactive {
+		return nil
+	}
+
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	var repoToCreate ghrepo.Interface
+	cfg, err := opts.Config()
+	if err != nil {
+		return err
+	}
+
+	host, _ := cfg.Authentication().DefaultHost()
+
+	opts.Name, opts.Description, opts.Visibility, err = interactiveRepoInfo(httpClient, host, opts.Prompter, "")
+	if err != nil {
+		return err
+	}
+
+	if !strings.Contains(opts.Name, "/") {
+		username, _, err := userAndOrgs(httpClient, host)
+		if err != nil {
+			return err
+		}
+		opts.Name = fmt.Sprintf("%s/%s", username, opts.Name)
+	}
+	repoToCreate, err = ghrepo.FromFullName(opts.Name)
+	if err != nil {
+		return fmt.Errorf("argument error: %w", err)
+	}
+
+	input := repoCreateInput{
+		Name:               repoToCreate.RepoName(),
+		Visibility:         opts.Visibility,
+		OwnerLogin:         repoToCreate.RepoOwner(),
+		TeamSlug:           opts.Team,
+		Description:        opts.Description,
+		HomepageURL:        opts.Homepage,
+		HasIssuesEnabled:   !opts.DisableIssues,
+		HasWikiEnabled:     !opts.DisableWiki,
+		GitIgnoreTemplate:  opts.GitIgnoreTemplate,
+		LicenseTemplate:    opts.LicenseTemplate,
+		IncludeAllBranches: opts.IncludeAllBranches,
+		InitReadme:         opts.AddReadme,
+	}
+
+	templateRepo, err := interactiveRepoTemplate(httpClient, host, repoToCreate.RepoOwner(), opts.Prompter)
+	if err != nil {
+		return err
+	}
+	input.TemplateRepositoryID = templateRepo.ID
+	templateRepoMainBranch := templateRepo.DefaultBranchRef.Name
+
+	targetRepo := shared.NormalizeRepoName(opts.Name)
+	if idx := strings.IndexRune(opts.Name, '/'); idx > 0 {
+		targetRepo = opts.Name[0:idx+1] + shared.NormalizeRepoName(opts.Name[idx+1:])
+	}
+	confirmed, err := opts.Prompter.Confirm(fmt.Sprintf(`This will create "%s" as a %s repository on GitHub. Continue?`, targetRepo, strings.ToLower(opts.Visibility)), true)
+	if err != nil {
+		return err
+	} else if !confirmed {
+		return cmdutil.CancelError
+	}
+
+	repo, err := repoCreate(httpClient, repoToCreate.RepoHost(), input)
+	if err != nil {
+		return err
+	}
+
+	cs := opts.IO.ColorScheme()
+	isTTY := opts.IO.IsStdoutTTY()
+	if isTTY {
```

Comment:
You are right. This conditional branch is not necessary since it is determined that `opts.IO.IsStdoutTTY()` is true when the program is in interactive mode.